cardano-db-sync/app/cardano-db-sync.hs

```diff
@@ -188,15 +190,24 @@ pHasCache =
         <> Opt.help "Disables the db-sync caches. Reduces memory usage but it takes longer to sync."
     )
 
-pUseLedger :: Parser Bool
-pUseLedger =
+pHasLedger :: Parser Bool
```

Comment:
Could this be renamed to `ShouldUseLedgerState`? 

Just it's a little confusing when there is both `HasLedger` + `ShouldUseLedger` 🤔

Reply:
I'm just saying `HasLedger` makes it confusing in it's naming, putting the word `State` somewhere with it makes it clearer which flag it came from.

Another confusion is that if `HasLedger = False` and `ShouldUseLedger = True`. How can one "use the ledger" if it doesn't "have one".

Reply:
> I'm just saying HasLedger makes it confusing in it's naming, putting the word State somewhere with it makes it clearer which flag it came from.

It's quite common to use `Ledger` in code meaning `LedgerState`. Also the flags don't have the word state, ie `--disable-ledger`, `--dont-use-ledger`

> Another confusion is that if HasLedger = False and ShouldUseLedger = True. How can one "use the ledger" if it doesn't "have one".

I think this is not possible, since 
```
useLedger = enpHasLedger snp && enpShouldUseLedger snp && not (enpOnlyUTxO snp)
```
useLedger is True only if hasLedger is True

Reply:
ok cool no worries 👍